lib/topics/state_and_behaviour.rb

```diff
@@ -2,4 +2,30 @@
 
 # exercise state and behaviour
 module StateAndBehaviour
+  # exercise car model
+  class Car
+    attr_reader :year, :color, :model, :current_speed
+    
+    def self.default_car
+      car_attributes = {}
+      new(car_attributes)
+    end
+
+    def initialize(car_attributes)
+      raise ArgumentError unless car_attributes.is_a?(Hash)
```

Comment:
what about a universal and reusable method for type checking? Sounds interesting